### ISSUE TITLE
German Translation breaks Layout + two more Suggestion

### ISSUE THREAD

**Issue by lunainvictum** [FreeTubeApp/FreeTube#826]
Hey @ all!

First: GREAT piece of Software. I am so happy that i found FreeTube.

Why i searched for it? Because i use already Newpipe one my Phone, because i hate the whole "Watch this Video, Suggestion here, Crap here, watch this"-Thing. 

The one (or two, depends how you count it) reason i use youtube, is education and music - for nothing more. 

And sure i can ignore everything what is displayed on the Startpage of Youtube. But it bothers me regardless and makes me sometimes angry (thanks for you care. im in stable health :) ). I tried years ago few extension for my Browser with them i could remove all Channels i dont want. But this is a tidious work. It exist more channels than i have brain cells left after all the "Internetkacke" (how we call it in germany. Internetshit maybe is the right translation). 

And i dont found a Addon, to use a Whitelistmethod instead of Blacklist. 

But then i found FreeTube! You cant believe how happy i am that i found that. 

Its bothers me a little that 4k Playback isnt possible. But i read that is a restriction from Youtube. So every fine.

I dont know if i could help with this problem, but for this i wrote a solution on linux shell (bash script). Not worth to share, because i used for my Mediapc (linux without desktop, only with X server and mpv).

What it did: It simply watched a document. And if the document was edited (from my smartphone with editor), it copied the youtube link. Downloaded it on maximal Quality, and than played it with mpv. The Reason because this is possible for me, is because i have a 1gbs/1gbs Internetconnection at home. So i have only to wait 1-5 Secounds for the typical youtube content i watch. 

And that would be great to implement such option to freetube wich must be opt-in. I saw there is already a download button (wich i dont tried yet). Then i could watch my videos again in 4k :-) 

But like i mentioned, that is only nice to have.

What really would be great is the possibility to use tabs or something. Or background play. Because i have often scenarios that i only listen to music, and want to play the next song without pause. So what i do with a browser, is to open two tabs, play one song, in the other tab i already search the next song i like to hear and then wait until the first song is over and press play on the secound tab. 

Now i use the picture in picture mode as workaround in freetube. 

And there i found a bug too. If you open a video and watch it in pip mode, then search another and click on that, and if you by accident click on the pip on the little pip window again, that video dont comes back, instead the one starts that you searched and waited on to start. I hope i could it explain it. My english is very bad. 

Lets going back to the Title topic:

<img width="1061" alt="2020-12-03 04_32_38-New Issue · FreeTubeApp_FreeTube - Chromium" src="https://user-images.githubusercontent.com/50776501/100960484-a6499d80-3520-11eb-9897-33e1f3637467.png">

This breaks the layout. I already found the language files, but there are .pak files. if i open these with notepad++, i have only chinese (i think, symbols i cant read) letters/symbols. 

I wanted to change "Autmatische Wiedergabe" simply to Autoplay or something another shorter. 

So on, my last suggestion: A option to hide/edit/remove the sidebar on the left side. It would be a pleasure if i could remove the Trending, Most Popular, Playlists Buttons. But because Freetube is so nice, and well programmed, that it everytimes opens with the subscription windows, it isnt so important. only a cosmetic thing (to prevent that if im drunk af and cant get over the tempation to watch 23 year old mandy with blond haires to show what nice hairlotions she bought 41021 kilometers away in a little green vegan store made from plants that life on free mind and get only chopped from a guy who is a educated naturewhisperer in hemp clothes and wood sandals).

Long Story short: thanks for this great programm!

Edit: and please, please dont be angry to me and forgive my horrible english. im a rly old guy wich lives in germany.

**Comment by GilgusMaximus** [comment]
Hi, great to hear you enjoy the application.
I'm gonna have to unwrap a lot here and just gonna say ahead so my answer's gonna be a bit longer as well.


> Its bothers me a little that 4k Playback isnt possible. But i read that is a restriction from Youtube. So every fine.

You can play 4K videos, but only the ones that have 4K in mp4 format available. This is unfortunately a current limitation of the vhs plugin of VideosJS. In the future they want to support webm format as well, so all videos should work once that is available.
So just search 4k and test a few videos. Some work some not.


> What really would be great is the possibility to use tabs or something

Please refer to #333 



> Or background play

You can play videos in Audio Only format. But because of the missing tabs you cannot search for other things at the same time in FreeTube.



> hat video dont comes back

Please refer to #624 



> This breaks the layout.

Yes we know that. A lot of languages break the layout. e.g. Finnish. We play to implement a responsive way at some point in time when all the major features are available. Because then we have to fix the whole layout once and not every time a new feature comes in.


>  wanted to change "Autmatische Wiedergabe" simply to Autoplay or something another shorter.

You can either go to our [Weblate](https://hosted.weblate.org/engage/free-tube/) page where all the translations are made and suggest a change, which in this case is unlikely to occur because the translation is correct. For your case it would be best to clone the source code via git and change the file /static/locale/de-DE.yaml where the German translation is stored in. And then you build the application yourself as described in our [wiki](https://github.com/FreeTubeApp/FreeTube/wiki/Environment-Setup-and-Building-from-Source). Then whenever something changes here you can pull the data and build again to have your version.


> A option to hide/edit/remove the sidebar on the left side. It would be a pleasure if i could remove the Trending, Most Popular, Playlists Buttons

You can go to the Settings and then to Distraction Free Mode (in German: Einstellungen für ablenkungsfreien Modus) and deactivate Trends and popular videos with the two buttons (Trends verbergen und Beliebte Videos verbergen).


Thanks again for the positive feedback. If there is anything left feel free to write in here again, otherwise if nothing is left open I'd close this Issue at some point in time : )

Best Regards,
GilgusMaximus

**Comment by lunainvictum** [comment]
`You can go to the Settings and then to Distraction Free Mode (in German: Einstellungen für ablenkungsfreien Modus) and deactivate Trends and popular videos with the two buttons (Trends verbergen und Beliebte Videos verbergen).`

wow im blind! that is exactly that! Great! 

What i want to mention: If you hide the recommended Videos, the layout is broken too. 
![2020-12-03 19_11_59-German Translation breaks Layout + two more Suggestion · Issue #826 · FreeTubeAp](https://user-images.githubusercontent.com/50776501/101070365-8b683f00-359b-11eb-9ae6-230a35267547.png)

This could be maybe dynamic (same way if you resize the window self). 

`You can either go to our Weblate page where all the translations are made and suggest a change, which in this case is unlikely to occur because the translation is correct. For your case it would be best to clone the source code via git and change the file /static/locale/de-DE.yaml where the German translation is stored in. And then you build the application yourself as described in our wiki. Then whenever something changes here you can pull the data and build again to have your version.`

Yeah i already found that, and trying it next week out. Would it maybe not easier to use xml files for example? 

But for now, that problem is fixed anyway, because i disabled recommended Videos :-)

And i dont know if here already exist a thread about it, but sometimes i get "Local API SyntaxError: Unexpected end of JSON input" or "Invidious API [object Object]" error.

**Comment by GilgusMaximus** [comment]
For the issue with the sidebar please open a new issue if no other exists yet. This makes it easier to identify for others and we can potentially pull in the contributor who implemented the distraction free mode.

We use yaml because PrestoN at one point decided to use it and changing every yaml file doesn't make sense at this point anymore.

For the error, it is relevant when this occurs. On search, on playback..etc. You can search for most error messages within all issues. Additionally, Invidious problems need to be resolved over at Invidious. If no other issue with the local API error exists you can create a new issue. Although I'd always first recommend downloading the newest nightly version to check whether this persists.

Because no other questions were raised I'm closing this issue. If more help is required please open 1 issue per problem / discussion topic to keep the issue page easily navigable and searchable.
